lib/hypervisor/hv_kvm/monitor.py

```diff
@@ -538,9 +538,15 @@ def HotAddDisk(self, disk, devid, uri, drive_add_fn=None):
 
     """
     if os.path.exists(uri):
-      fd = os.open(uri, os.O_RDWR)
-      fdset = self._AddFd(fd)
-      os.close(fd)
+      # For QEMU's dynamic auto-read-only feature to work (on by default since
+      # QEMU 4.0), we need two file descriptors, one in read-write and one in
+      # read-only mode.
+      fd_rw = os.open(uri, os.O_RDWR)
+      fd_ro = os.open(uri, os.O_RDONLY)
+      fdset = self._AddFd(fd_rw)
+      self._AddFd(fd_ro, fdset)
```

Comment:
QQ: is there a need for ordering here (rw before ro), or not? The man page for `qemu-system` shows this ordering, but it's not clear if any order is accepted or not.

Reply:
No need for ordering, I verified this against QEMU code: it merely iterates over all descriptors in the fdset until it finds one with matching flags.